src/linearmixedmodel.jl

```diff
@@ -474,8 +472,10 @@ end
     issingular(m::LinearMixedModel, θ=m.θ)
 
 Test whether the model `m` is singular if the parameter vector is `θ`.
+
+Equality comparisons are used b/c small non-negative θ values are replaced by 0 in `fit!`.
 """
-issingular(m::LinearMixedModel, θ=m.θ) = any(isapprox.(lowerbd(m), θ))
+issingular(m::LinearMixedModel, θ=m.θ) = any(lowerbd(m) .== θ)
```

Comment:
I get perverse pleasure in warranted exact comparisons in floating point. :)

Reply:
I am making the assumption that the value of θ is the result of a call to `fit!`, which should be the case. Nevertheless it is a hidden assumption on the validity of the method but I don't think it is necessary to be too pedantic about this one.

Reply:
I completely agree and no snark was intended. I think the exact comparison here is actually nice because it means we only have one set of thresholds (those in `fit!`) for numerical singularity.